src/main/java/com/iota/iri/service/API.java

```diff
@@ -804,35 +718,26 @@ public static void incEllapsedTime_PoW(long ellapsedTime) {
         return elements;
     }
 
-    private AbstractResponse addNeighborsStatement(final List<String> uris) throws URISyntaxException {
-
+    private AbstractResponse addNeighborsStatement(final List<String> uris) {
         int numberOfAddedNeighbors = 0;
-        for (final String uriString : uris) {
-            final URI uri = new URI(uriString);
 
-            if ("udp".equals(uri.getScheme()) || "tcp".equals(uri.getScheme())) {
```

Comment:
it seems as if this part of the refactor got a little out of the scope of the Pull Request's intention.
But I see how you factored `if ("udp".equals(uri.getScheme()) || "tcp".equals(uri.getScheme())) {` into a function to avoid code repetition.